dbt/adapters/spark/impl.py

```diff
@@ -93,53 +120,97 @@ def drop_relation(self, relation, model_name=None):
             connection_name=model_name
         )
 
-    def get_catalog(self, manifest):
-        schemas = manifest.get_used_schemas()
-
-        column_names = (
-            'table_database',
-            'table_schema',
-            'table_name',
-            'table_type',
-            'table_comment',
-            'table_owner',
-            'column_name',
-            'column_index',
-            'column_type',
-            'column_comment',
+    @staticmethod
+    def _parse_relation(relation: Relation,
+                        table_columns: List[Column],
+                        rel_type: str,
+                        properties: Dict[str, str] = None) -> List[dict]:
+        properties = properties or {}
+        statistics = {}
+        table_owner_key = 'Owner'
+
+        # First check if it is present in the properties
+        table_owner = properties.get(table_owner_key)
+
+        found_detailed_table_marker = False
+        for column in table_columns:
+            if column.name == '# Detailed Table Information':
+                found_detailed_table_marker = True
+
+            # In case there is another column with the name Owner
+            if not found_detailed_table_marker:
+                continue
+
+            if not table_owner and column.name == table_owner_key:
+                table_owner = column.data_type
+
+            if column.name == 'Statistics':
```

Comment:
Hi @Fokko you seem to be missing a test for parsing the Statistics

Reply:
Great catch Niels. I have a test on my fork: https://github.com/fishtown-analytics/dbt-spark/compare/master...Fokko:master#diff-cb0ffa1fa3373b03d243ccd85a226cc6R404

However, these tests are all mocked, something that I don't really like. Especially with Spark 3.0 coming up. It is better to directly talk to Spark, and don't rely on mock that might not reflect the actual interface. It would be great to update the Spark version when a new one gets released.

I'm working on a docker based CI for running the tests. This is a docker container with Hadoop, Hive, and Spark. This one will expose the HTTP 10001 port for doing the queries against Spark. With docker, we can test against 2.4.x which is the LTS release, and 3.0 which is currently in preview. This requires Hive as well for storing the metadata. Currently, I'm occupied with another project, and continuing the work on this end of next week.